source/common/upstream/health_checker_impl.cc

```diff
@@ -725,13 +732,12 @@ void GrpcHealthCheckerImpl::GrpcActiveHealthCheckSession::onResetStream(Http::St
 void GrpcHealthCheckerImpl::GrpcActiveHealthCheckSession::onGoAway() {
   ENVOY_CONN_LOG(debug, "connection going away health_flags={}", *client_,
                  HostUtility::healthFlagsToString(*host_));
-  // Even if we have active health check probe, fail it on GOAWAY and schedule new one.
+  // If we have an active health check probe, allow it to complete before closing the connection.
   if (request_encoder_) {
-    handleFailure(envoy::data::core::v3::NETWORK);
-    expect_reset_ = true;
-    request_encoder_->getStream().resetStream(Http::StreamResetReason::LocalReset);
+    received_goaway_ = true;
```

Comment:
A one liner comment here about when this connection would eventually get closed would be nice. IIUC it's either on the next timeout or when the remote closes it?

Reply:
I'll add a comment. It would be closed when the active check completes (`onRpcComplete`) or other conditions like `onResetStream` and `onTimeout`.